.github/workflows/build-extension-catalog.yml

```diff
@@ -1,20 +1,34 @@
-name: Build and release container to registry
+name: Build and release Extension Catalog Image to registry
 
 on:
-  push:
-    branches:
-      - main
-
-env:
-  REGISTRY: ghcr.io
-  IMAGE_NAME: ${{ github.repository }}
+  workflow_call:
+    inputs:
+      registry_target:
+        required: true
+        type: string
+      registry_user:
+        required: true
+        type: string
+      tagged_release:
+        required: false
+        type: string
+    secrets:
+      registry_token:
+        required: true
 
 jobs:
   build:
     name: Build container image
     if: github.ref == 'refs/heads/main' && github.event_name != 'pull_request'
     runs-on: ubuntu-latest
-    permissions: write-all
+    permissions:
```

Comment:
do we need all these permissions? have you checked if there are some that aren't need? a bit nitpicky, I know, but we should only use the essential ones, imo

Reply:
I've updated the permissions to only include the ones needed, previously we were requiring `write-all` which is certainly not necessary. I also updated the documentation to explain the necessary requirements.

The Github documentation around permissions can be a bit [verbose](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28) - but in short:
- `actions` requires write to [cancel a workflow](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#cancel-a-workflow-run)
- `contents` requires read for `actions/checkout` and requires write to `put` the contents of the built extension charts [into a branch](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents)
- `deployments` requires write when [deploying  `gh-pages`](https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment)
- `packages` requires write when a catalog image is created to [create the package](https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#publishing-a-package-using-an-action)
- `pages` requires write to [request and create page builds](https://docs.github.com/en/rest/pages/pages?apiVersion=2022-11-28#request-a-github-pages-build) for the deployment